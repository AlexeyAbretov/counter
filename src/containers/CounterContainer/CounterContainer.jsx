import React, { useContext } from 'react';

import { Counter } from '@components';
import { CounterContext } from '@contexts';

export const CounterContainer = () => {
    const {
        state
    } = useContext(CounterContext)

    return (
        <Counter title={state.counterValue} />
    )
}