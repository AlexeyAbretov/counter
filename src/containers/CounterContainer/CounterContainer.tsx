import React from 'react';

import { Counter, CounterType } from '@components';
import { useSelector } from 'react-redux';
import { getCounterContanerPropsSelector } from '@store';

export const CounterContainer: React.FC = (): JSX.Element => {
    const { value, type } = useSelector(getCounterContanerPropsSelector);
    
    return (
        <Counter
            title={String(value)}
            type={type as CounterType}
        />
    )
}