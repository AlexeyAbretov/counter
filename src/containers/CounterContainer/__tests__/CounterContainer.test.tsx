/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import { ThemeProvider } from 'styled-components';
import React from 'react';
import { theme } from '@theme';
import { CounterContext, CounterStore } from '@store';
import { render } from '@testing-library/react';
import { CounterContainer } from '../CounterContainer';

const getUi = (store: CounterStore): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <CounterContext.Provider value={store}>
        <CounterContainer />
      </CounterContext.Provider>
    </ThemeProvider>
  );
};

describe('CounterContainer', () => {
  it('should render zero', () => {
    const { container } = render(getUi(new CounterStore()));

    expect(container).toMatchSnapshot();
  });

  it('should render lesszero', () => {
    const { container } = render(getUi(new CounterStore(-1)));

    expect(container).toMatchSnapshot();
  });

  it('should render abovezero', () => {
    const { container } = render(getUi(new CounterStore(1)));

    expect(container).toMatchSnapshot();
  });
});
