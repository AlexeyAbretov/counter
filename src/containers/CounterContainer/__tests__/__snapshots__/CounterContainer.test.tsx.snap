// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CounterContainer > should render abovezero 1`] = `
<div>
  <div
    class="sc-gswNZR gxVSuE"
    data-testid="counter"
    type="abovezero"
  >
    1
  </div>
</div>
`;

exports[`CounterContainer > should render lesszero 1`] = `
<div>
  <div
    class="sc-gswNZR bZfoa-d"
    data-testid="counter"
    type="lesszero"
  >
    -1
  </div>
</div>
`;

exports[`CounterContainer > should render zero 1`] = `
<div>
  <div
    class="sc-gswNZR cneBUb"
    data-testid="counter"
    type="zero"
  >
    0
  </div>
</div>
`;
